internal/declarative/v2/reconciler.go

```diff
@@ -114,6 +117,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return r.ssaStatus(ctx, obj, metrics.ManifestInit)
 	}
 
+	logf.FromContext(ctx).Info("Before setting the Mandatory Module Metrics")
+	if obj.GetLabels()[shared.IsMandatoryModule] == strconv.FormatBool(true) {
```

Comment:
This shouldn't happen but just in case can you add `if object.GetLabels() != nil ` to the check as well?